src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { HtmlDiv } from '../../../../reset';
+import { windowAvailable } from '../../../../utils/common';
+import { AutoId } from '../../../../utils/AutoId';
+import { Debounce } from '../../../utils/Debounce/Debounce';
+import { Button } from '../../../Button/Button';
+import { Chip } from '../../../Chip';
+import { Popover } from '../../../Popover/Popover';
+import { FilterInput, FilterInputStatus } from '../../FilterInput/FilterInput';
+import { ComboboxItemList } from '../ComboboxItemList/ComboboxItemList';
+import { ComboboxItem } from '../ComboboxItem/ComboboxItem';
+import { ComboboxEmptyItem } from '../ComboboxEmptyItem/ComboboxEmptyItem';
+import { ChipList } from '../ChipList/ChipList';
+import { baseStyles } from './Combobox.baseStyles';
+
+const baseClassName = 'fi-combobox';
+const comboboxClassNames = {
+  wrapper: `${baseClassName}_wrapper`,
+  open: `${baseClassName}--open`,
+  removeAllButton: `${baseClassName}_removeAllButton`,
+  error: `${baseClassName}--error`,
+};
+
+export interface ComboboxData {
+  /** Unique label that will be shown on combobox item and used on filter */
+  labelText: string;
+  /** Using labelText if chipText is not given */
+  chipText?: string;
+  /** Item selection disabled for the user */
+  disabled?: boolean;
+  /** Unique id to identify the item */
+  uniqueItemId: string;
+}
+
+export interface ComboboxProps<T extends ComboboxData> {
+  /** Combobox container div class name for custom styling. */
+  className?: string;
+  /** Items for the combobox */
+  items: Array<T & ComboboxData>;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Label */
+  labelText: string;
+  /** Event that is sent when item selections change */
+  onItemSelectionsChange?: (selectedItems: Array<T>) => void;
+  /** Show chip list */
+  chipListVisible?: boolean;
+  /** Chip action label */
+  chipActionLabel?: string;
+  /** Label for remove button. If it is given, button will be shown. */
+  removeAllButtonLabel?: string;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Label to show when no items to show, e.g filtered all out */
+  emptyItemsLabel: string;
```

Comment:
`noItemsLabel`, `noResultsLabel` or even -Text rather than label?

Reply:
Maybe `noItemsText` would be good one?

Reply:
Text is better than label as it really is text in this case and will be rendered in "different" context.

Reply:
This is now changed to `noItemsText`.